tools/swift-reflection-dump/swift-reflection-dump.cpp

```diff
@@ -120,6 +137,93 @@ static bool needToRelocate(SectionRef S) {
   return true;
 }
 
+static section_iterator findSectionByName(StringRef Name, const ObjectFile &O) {
+  auto Sections = O.sections();
+  SmallVector<section_iterator, 1> FoundSections;
+  std::copy_if(Sections.begin(), Sections.end(),
+               std::back_inserter(FoundSections), [Name](SectionRef S) {
+                 StringRef N;
+                 if (auto EC = S.getName(N))
+                   reportError(EC);
+                 return N == Name;
+               });
+  if (FoundSections.size() != 1)
+    return Sections.end();
+  return FoundSections.front();
+}
+
+template <class ELFObj,
+          typename std::enable_if<
+              std::is_base_of<ELFObjectFileBase, ELFObj>::value, int>::type = 0>
+const typename ELFObj::Elf_Rela *findRelaForOffset(uint64_t Offset,
+                                                   const ELFObj &O) {
+  for (const SectionRef &S :
+       static_cast<const ObjectFile *>(&O)->dynamic_relocation_sections()) {
+    const auto *Sec = O.getSection(S.getRawDataRefImpl());
+    if (Sec->sh_type != llvm::ELF::SHT_RELA)
+      continue;
+    for (const RelocationRef &R : S.relocations()) {
+      const auto *Rela = O.getRela(R.getRawDataRefImpl());
+      if (Rela->r_offset == Offset &&
+          Rela->getType(false /* isMips64EL */) == llvm::ELF::R_X86_64_RELATIVE)
```

Comment:
Do we need to worry about other RELA relocations in ELF?  Or is this for the weird encoding of the x86_64 RELA in the PSABI?

Reply:
`MetadataSections` probably should use `RelativePointer`s. Aside from making it smaller and conster, that would save us having to try to fix its pointers up here.

Reply:
heh, those things (including the symbols __start_*, __stop_*) are generated by the linker, not the compiler,
maybe I'm missing something, but I don't see how RelativePointer can be used here.
Another concern regarding your change to swift-reflection-dump - I need to take a closer look at those commits if you don't mind,
so basically because of dynamic relocations we had to create a new memory buffer with the content of the object file, so that we can modify it (to emulate the loading of the binary into memory).